profile/profile.go

```diff
@@ -48,6 +48,8 @@ type Custom struct {
 	DisableIntendedRecipients bool
 	// AllowWeakRSA is a flag to disable checks for weak rsa keys.
 	AllowWeakRSA bool
+	// AllowSingingKeyInDecryption is a flag to enable to decrypt with signing keys for compatibility reasons.
+	AllowSingingKeyInDecryption bool
```

Comment:
Maybe same thing here?
```suggestion
	// InsecureAllowDecryptionWithSigningKeys is a flag to enable to decrypt with signing keys for compatibility reasons.
	InsecureAllowDecryptionWithSigningKeys bool
```
(arguably, `AllowWeakRSA` should also be called `InsecureAllowWeakRSA`, but that's an orthogonal point..)